packages/core/src/controllers/ChainController.ts

```diff
@@ -227,27 +222,35 @@ export const ChainController = {
     )
   },
 
-  setActiveChain(
+  setActiveNamespace(
     chain: ChainNamespace | undefined,
     caipNetwork?: NetworkControllerState['caipNetwork']
   ) {
-    const newAdapter = chain ? state.chains.get(chain) : undefined
-
-    if (newAdapter && newAdapter.chainNamespace !== state.activeChain) {
-      state.activeChain = newAdapter.chainNamespace
-      state.activeCaipAddress = newAdapter.accountState?.caipAddress
-      state.activeCaipNetwork = caipNetwork
-
-      if (!newAdapter.accountState) {
-        this.resetAccount(newAdapter.chainNamespace)
+    if (caipNetwork?.chainNamespace) {
+      const newAdapter = chain ? state.chains.get(caipNetwork.chainNamespace) : undefined
+      const newNamespace = newAdapter?.chainNamespace !== state.activeChain
+
+      if (newAdapter && newNamespace) {
+        state.activeChain = newAdapter.chainNamespace
+        state.activeCaipNetwork = caipNetwork
+        state.activeCaipAddress = newAdapter.accountState?.caipAddress
+        SafeLocalStorage.setItem(SafeLocalStorageKeys.ACTIVE_CAIP_NETWORK_ID, caipNetwork.id)
+
+        NetworkController.replaceState(newAdapter.networkState)
+        AccountController.replaceState(newAdapter.accountState)
```

Comment:
we should def get rid of replace state everywhere, but diff pr for sure

Reply:
Exactly, noted